### PR TITLE
Homework20: Marie Ambrazhei 

### PR DESCRIPTION
**Pull request by MarieAmbrazhei**

### CODE REVIEW

File: homework20/test_library_fuctionality.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        del self.book3
+
+        del self.user_1
+        del self.user_2
+
+    def test_take_book_positive(self):
+        """Test taking a book that is available."""
+        log.info('User tries to take book with 134527 isbn')
+        self.assertTrue(self.user_1.take_book('134527'))
+        self.assertIn('134527', User.taken_books)
```
Comment (Stml89) [inline homework20/test_library_fuctionality.py:49]: Add something like: `Successfully verified <...>`

File: requirements.txt
Comment (Stml89) [inline requirements.txt:1]: do you need all of those libraries?

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework20/test_library_fuctionality.py (hunk 1 above)
Comment (MarieAmbrazhei) [reply homework20/test_library_fuctionality.py:49]: added

File: requirements.txt
Comment (Stml89) [reply requirements.txt:1]: it fetches all libs that you have installed

Comment (MarieAmbrazhei) [reply requirements.txt:1]: got it. done!

File: homework20/test_library_fuctionality.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        Book.books = {}
+        User.users_books = {}
+
+        self.book1 = Book("Learning Python", "134527", 1200,
+                          "Mark Lutz")
+        self.book2 = Book("Learn Python 3 the Hard Way", "234569",
+                          1000,
+                          "Zed A. Shaw")
+        self.book3 = Book("Think Python", "239874",
+                          900, "Allen Downey ")
```
Comment (Stml89) [inline homework20/test_library_fuctionality.py:30]: You have created those books but not using them

File: homework20/test_library_fuctionality.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        log.info(f'Successfully verified: User took book with  {isbn} isbn')
+
+    def test_take_book_already_taken(self):
+        """Test that a user cannot take a book that is already taken."""
+        isbn = '134527'
+        log.info('User tries to take an already taken book with 134527 isbn')
+        self.user_1.take_book('134527')
+        self.assertFalse(self.user_2.take_book('134527'))
+        log.info(f'Successfully verified: book with {isbn} is unavailable,'
+                 f' additional info: {self.books.get(isbn)}')
```
Comment (Stml89) [inline homework20/test_library_fuctionality.py:65]:
```suggestion
        """Test that a user cannot take a book that is already taken."""
        log.info('User tries to take an already taken book with {self.book1.isbn} isbn')
        self.user_1.take_book({self.book1.isbn})
        self.assertFalse(self.user_2.take_book(self.book1.isbn))
        log.info(f'Successfully verified: book with {{self.book1.isbn}} is unavailable,'
                 f' additional info: {self.books.get({self.book1.isbn})}')
```

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework20/test_library_fuctionality.py (hunk 3 above)
Comment (MarieAmbrazhei) [reply homework20/test_library_fuctionality.py]: done

**Review by Stml89 — APPROVED** [review]

**Pull request merged**